aiida/orm/utils/mixins.py

```diff
@@ -105,13 +109,47 @@ def _set_function_starting_line_number(self, function_starting_line_number):
         """
         self.set_attribute(self.FUNCTION_STARTING_LINE_KEY, function_starting_line_number)
 
+    @property
+    def function_number_of_lines(self):
+        """Return the number of lines of the wrapped function in its source file.
+
+        :returns: the number of lines or None
+        """
+        return self.get_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, -1)
+
+    def _set_function_number_of_lines(self, function_number_of_lines):
+        """Set the number of lines of the wrapped function in its source file.
+
+        :param function_number_of_lines: the number of lines
+        """
+        self.set_attribute(self.FUNCTION_NUMBER_OF_LINES_KEY, function_number_of_lines)
+
     def get_function_source_code(self):
-        """Return the absolute path to the source file in the repository.
+        """Return the content of the entire source file that contains the process function.
 
-        :returns: the absolute path of the source file in the repository, or None if it does not exist
+        :returns: the content of the source file
         """
+        warnings.warn(
+            'in the future this will return only the source of the function and'
+            ' please use `get_function_source_file` to get'
+            ' the entire content of the file instead', AiidaDeprecationWarning
+        )
         return self.get_object_content(self.FUNCTION_SOURCE_FILE_PATH)
 
+    def get_function_source(self):
+        """Return the source code of the function and only of the function.
```

Comment:
If we are doing the deprecation path, I would add to this docstring that this is a temporary method.

I'm also wondering if we should add a test for this or not, considering that it is going to be temporary. I guess it would be the same that adding a unit test for the warning. @sphuber would it be a good idea to have a unit test file with all the deprecation warnings, considering it would have to be explicitly updated when removing the deprecations? Somehow it sounds both a bit insane but not a bad idea at the same time, as a way to keep track of all deprecations in a single place...

Reply:
The three relevant functions are closely packed in the code. So I think the source code itself with its docstring makes enough explanation for how to do the future migration. Let's see if @sphuber has any better ideas.